jujugui/static/gui/src/app/components/button-dropdown/button-dropdown.js

```diff
@@ -0,0 +1,142 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+
+'use strict';
+
+/**
+  Creates a component that has an icon, which when clicked, opens a dropdown
+  menu with list items.
+*/
+class ButtonDropdown extends React.Component {
+  constructor() {
```

Comment:
Please let's not change the signature of the constructor, let's accept and propagate props.